File: Yow.CoD.Finance.Domain/Contracts/LoanBankAccountChangedEvent.cs
namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class LoanBankAccountChangedEvent : Event
    {
        public LoanBankAccountChangedEvent(string bankAccountBsb, string bankAccountAccountNumber)
        {
            BankAccount = new BankAccount(bankAccountBsb, bankAccountAccountNumber);
        }

        public BankAccount BankAccount { get; }
    }
}

File: java/domain/src/main/java/com/leecampbell/cod/domain/contracts/LoanBankAccountChangedEvent.java
package com.leecampbell.cod.domain.contracts;

public final class LoanBankAccountChangedEvent extends DomainEvent {
    private final BankAccount bankAccount;

    public LoanBankAccountChangedEvent(String bankAccountBsb, String bankAccountAccountNumber) {
        this.bankAccount = new BankAccount(bankAccountBsb, bankAccountAccountNumber);
    }

    public BankAccount getBankAccount() {
        return bankAccount;
    }
}